python/lsst/validate/drp/calcSrd.py

```diff
@@ -259,18 +267,225 @@ def computeWidths(array):
     Returns
     -------
     float, float
-        RMS and scaled inter-quartile range (IQR).  
-        
+        RMS and scaled inter-quartile range (IQR).
+
     Notes
     -----
-    We estimate the width of the histogram in two ways: 
-       using a simple RMS, 
+    We estimate the width of the histogram in two ways:
+       using a simple RMS,
        using the interquartile range (IQR)
     The IQR is scaled by the IQR/RMS ratio for a Gaussian such that it
-       if the array is Gaussian distributed, then the scaled IQR = RMS. 
+       if the array is Gaussian distributed, then the scaled IQR = RMS.
     """
     rmsSigma = math.sqrt(np.mean(array**2))
     iqrSigma = np.subtract.reduce(np.percentile(array, [75, 25])) / (scipy.stats.norm.ppf(0.75)*2)
     return rmsSigma, iqrSigma
 
 
+def sphDist(ra1, dec1, ra2, dec2):
+    """Calculate distance on the surface of a unit sphere.
+
+    Input and Output are in radians.
+
+    Notes
+    -----
+    Uses the Haversine formula to preserve accuracy at small angles.
+
+    Law of cosines approach doesn't work well for the typically very small
+    differences that we're look at here.
+    """
+    # Haversine
+    dra = ra1-ra2
+    ddec = dec1-dec2
+    a = np.square(np.sin(ddec/2)) + \
+        np.cos(dec1)*np.cos(dec2)*np.square(np.sin(dra/2))
+    dist = 2 * np.arcsin(np.sqrt(a))
+
+    # This is what the law of cosines would look like
+#    dist = np.arccos(np.sin(dec1)*np.sin(dec2) + np.cos(dec1)*np.cos(dec2)*np.cos(ra1 - ra2))
+
+    # Could use afwCoord.angularSeparation()
+    #  but (a) that hasn't been made accessible through the Python interface
+    #  and (b) I'm not sure that it would be faster than the numpy interface.
+    #    dist = afwCoord.angularSeparation(ra1-ra2, dec1-dec2, np.cos(dec1), np.cos(dec2))
+
+    return dist
+
+
+def matchVisitComputeDistance(visit_obj1, ra_obj1, dec_obj1,
+                              visit_obj2, ra_obj2, dec_obj2):
+    """Match visit_obj1 and visit_obj2 and calculate ra, dec distance for matches.
```

Comment:
It matches _visits_ for a pair of separated objects
and then reports their distance for each matching visit.

Clearly the docstring needs improvement.  I will work to clarify what this does.